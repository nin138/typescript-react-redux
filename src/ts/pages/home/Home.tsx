import * as React from 'react';
import { connect } from 'react-redux';
import { Action, Dispatch } from 'redux';
import styles from '../../../css/app.css';
import { AppState } from '../../store';
import { changeText, requestRandomText } from './actions';

interface Props {
  text: string;
  actions: ActionDispatcher;
}

class ActionDispatcher {
  constructor(private dispatch: Dispatch<Action<unknown>>) {}
  changeText = (text: string) => {
    this.dispatch(changeText(text));
  };
  requestRandomText = () => {
    this.dispatch(requestRandomText());
  };
}

const Home = (props: Props) => {
  const handleChange = (e: React.ChangeEvent<HTMLInputElement>) => props.actions.changeText(e.target.value);
  return (
    <section className={styles.app}>
      <h1>Home</h1>
      <img src={'./images/logo.svg'} alt={'logo'} height={16} width={16} />
      {props.text || 'hello, world!'}
      <br />
      <input type={'text'} value={props.text} onChange={handleChange} />
      <br />
      <button onClick={props.actions.requestRandomText}>async random text</button>
    </section>
  );
};

export const HomeContainer = connect(
  (state: AppState) => state.home,
  dispatch => ({ actions: new ActionDispatcher(dispatch) })
)(Home);
