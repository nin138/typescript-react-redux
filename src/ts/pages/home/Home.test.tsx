import * as React from 'react';
import renderer from 'react-test-renderer';
import { Home } from './Home';

it('Home component', () => {
  expect(renderer.create(<Home text={'XD'} actions={{} as any} isWaiting={true} />).toJSON()).toMatchSnapshot();

  expect(renderer.create(<Home text={'(^^)'} actions={{} as any} isWaiting={false} />).toJSON()).toMatchSnapshot();
});
