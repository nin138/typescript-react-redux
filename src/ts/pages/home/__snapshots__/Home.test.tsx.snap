// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home component 1`] = `
<section
  className="app"
>
  <h1>
    Home
  </h1>
  <img
    alt="logo"
    height={16}
    src="./images/logo.svg"
    width={16}
  />
  XD
  <br />
  <input
    onChange={[Function]}
    type="text"
    value="XD"
  />
  <br />
  <button>
    async random text
  </button>
  <div
    style={
      Object {
        "display": "inline-block",
        "height": "1rem",
        "transform": "scale(.2)",
      }
    }
  >
    <div
      style={
        Object {
          "background": "#2196f3",
          "height": "30px",
          "width": "30px",
        }
      }
    />
    <div
      style={
        Object {
          "background": "#ff96f3",
          "height": "30px",
          "width": "30px",
        }
      }
    />
  </div>
</section>
`;

exports[`Home component 2`] = `
<section
  className="app"
>
  <h1>
    Home
  </h1>
  <img
    alt="logo"
    height={16}
    src="./images/logo.svg"
    width={16}
  />
  (^^)
  <br />
  <input
    onChange={[Function]}
    type="text"
    value="(^^)"
  />
  <br />
  <button>
    async random text
  </button>
</section>
`;
